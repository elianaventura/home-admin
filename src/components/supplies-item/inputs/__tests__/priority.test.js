import { render } from '@testing-library/react';
import Priority from '../priority';
import userEvent from '@testing-library/user-event';
import { screen } from '@testing-library/dom';

const props = {
  type: 'priority',
  field_name: 'priority',
  title: 'Prioridad:',
  number: 4,
  handleFieldChange: jest.fn(),
};

describe('Unit', () => {
  it('render passes snapshot', () => {
    const { container } = render(
      <Priority {...props} />
    );
 
    expect(container).toMatchSnapshot();
  });

  it('should change unit', async () => {
    userEvent.setup();
    const { getByTestId } = render(
      <Priority {...props} />
    );
 
    const input = getByTestId('priority-input');

    await userEvent.clear(input);
    await userEvent.type(input, '3');

    expect(input.value).toBe('3');

    expect(props.handleFieldChange).toHaveBeenCalledWith(
      props.field_name,
      '3',
    );
  });
});
