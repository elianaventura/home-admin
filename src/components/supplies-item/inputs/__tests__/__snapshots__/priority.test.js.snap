// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit render passes snapshot 1`] = `
<div>
  <div>
    <span
      class="priority-title"
    >
      Prioridad:
    </span>
    <input
      class="priority-input"
      data-testid="priority-input"
      value="4"
    />
  </div>
</div>
`;
